Add "Argon" unstable rate counter

Hiviexd opened the pull request:
https://github.com/user-attachments/assets/89202afc-0659-4cc0-aa09-1d0b5fb38533

---

Resolves #34591

Tried following the structure of the pp counter for this since they're on similar circumstances.

Hiviexd committed:
convert `UnstableRateCounter` to an abstract class and move triangles implementation to its own component

Hiviexd committed:
handle UR counter namespace change

Hiviexd committed:
add argon variant of UR counter

Hiviexd committed:
refactor tests to use both UR counter variants

Hiviexd committed:
fix warnings

Hiviexd committed:
address deserialisation tests

bdach commented on osu.Game/Screens/Play/HUD/UnstableRateCounter.cs:
```diff
@@ -67,65 +48,18 @@ private void updateDisplay()
         {
             base.Dispose(isDisposing);
 
-            if (scoreProcessor.IsNotNull())
-            {
```
please restore this check, it was there for a reason (ensures that this component is safe to dispose before dependency injection has taken place), it cannot be just removed like this for no reason

bdach commented on osu.Game/Screens/Play/HUD/UnstableRateCounter.cs:
```diff
@@ -3,48 +3,29 @@
-            valid.BindValueChanged(e =>
-                DrawableCount.FadeTo(e.NewValue ? 1 : alpha_when_invalid, 1000, Easing.OutQuint));
-        }
+        public virtual bool IsValid { get; set; }
```
not sure why this was changed *away* from being a bindable, only to then have inheritors of the class *override the setter* and essentially make it a bindable again. just make it a bindable again please

bdach commented on osu.Game/Screens/Play/HUD/ArgonUnstableRateCounter.cs:
```diff
@@ -0,0 +1,82 @@
+            return digitsRequired;
+        }
+
+        protected override LocalisableString FormatCount(int count) => count.ToString(@"D");
```
why override this?

bdach commented on osu.Game/Skinning/Triangles/TrianglesUnstableRateCounter.cs:
```diff
@@ -0,0 +1,85 @@
+            }
+        }
+
+        protected override LocalisableString FormatCount(int count) => count.ToString(@"D");
```
same question as above

Hiviexd committed:
add extra test

Hiviexd committed:
address review

bdach committed:
Fix initial alpha not applying correctly

Also fixes value change callbacks in BDL (very bad) and potential
lingering initial transforms.

bdach committed:
Fix test scene doing very weird things for (seemingly) no reason

bdach reviewed (approved):
@Hiviexd please check my commits at the end here for future reference:

- `BindValueChanged()` callbacks should never be registered in BDL due to threading concerns
- tests should never need to arbitrarily call `base.SetUpSteps()` in the middle of the test (and to my eye, they never did actually need to do that)

Hiviexd commented:
Good to know for future reference, thanks.
